strawberryfields/api_client.py

```diff
@@ -0,0 +1,584 @@
+
+        job = Job()
+        circuit = '''
+            name StateTeleportation
+            version 1.0
+            target gaussian (shots=1000)
+
+            complex alpha = 1+0.5j
+            Coherent(alpha) | 0
+            Squeezed(-4) | 1
+            Squeezed(4) | 2
+            BSgate(pi/4, 0) | (1, 2)
+            BSgate(pi/4, 0) | (0, 1)
+            MeasureX | 0
+            MeasureP | 1
+            Xgate(sqrt(2)*q0) | 2
+            Zgate(sqrt(2)*q1) | 2
+            MeasureHeterodyne() | 2
+        '''
+        job.manager.create(circuit=circuit)
+
+        job.id  # Returns the job ID that was generated by the server
+        job.reload()  # Fetches the latest job data from the server
+        job.status  # Prints the status of this job
+        job.result  # Returns a JobResult object
+        job.circuit  # Returns a JobCircuit object
+
+        job.result.reload()  # Reloads the JobResult object from the API
+
+        job.manager.get(1536)  # Fetches job 1536 from the server and updates the instance
+
+Classes
+-------
+
+.. autosummary::
+   APIClient
+   Resource
+   ResourceManager
+   Field
+   Job
+"""
+
+
+import urllib
+import json
+import warnings
+
+import dateutil.parser
+import requests
+
+from strawberryfields._dev import configuration
+
+
+def join_path(base_path, path):
+    """
+    Joins two paths, a base path and another path and returns a string.
+
+    Args:
+        base_path (str): The left side of the joined path.
+        path (str): The right side of the joined path.
+
+    Returns:
+        str: A joined path.
+    """
+    return urllib.parse.urljoin("{}/".format(base_path), path)
+
+
+class MethodNotSupportedException(TypeError):
+    """
+    Exception to be raised when a ResourceManager method is not supported for a
+    particular Resource.
+    """
+
+    pass
+
+
+class ObjectAlreadyCreatedException(TypeError):
+    """
+    Exception to be raised when an object has already been created but the user
+    is attempting to create it again.
+    """
+
+    pass
+
+
+class JobNotQueuedError(Exception):
+    """
+    Raised when a job is not successfully queued for whatever reason.
+    """
+
+    pass
+
+
+class JobExecutionError(Exception):
+    """
+    Raised when job execution failed and a job result does not exist.
+    """
+
+    pass
+
+
+class APIClient:
+    """
+    An object that allows the user to connect to the Xanadu Platform API.
+    """
+
+    USER_AGENT = "strawberryfields-api-client/0.1"
+
+    ALLOWED_HOSTNAMES = ["localhost", "localhost:8080", "platform.strawberryfields.ai"]
+
+    DEFAULT_HOSTNAME = "localhost"
+
+    ENV_KEY_PREFIX = "SF_API_"
+    ENV_AUTHENTICATION_TOKEN_KEY = "{}AUTHENTICATION_TOKEN".format(ENV_KEY_PREFIX)
+    ENV_API_HOSTNAME_KEY = "{}API_HOSTNAME".format(ENV_KEY_PREFIX)
+    ENV_USE_SSL_KEY = "{}USE_SSL".format(ENV_KEY_PREFIX)
+
+    DEFAULT_CONFIG = {"use_ssl": True, "hostname": DEFAULT_HOSTNAME, "authentication_token": None}
```

Comment:
Now that the `ENV_KEY_PREFIX` is defined in the `Configuration` class, and the default config defined in `configuration.DEFAULT_CONFIG`, I'm wondering if it makes sense to simply let the `Configuration()` class properly determine the defaults to avoid code duplication.

Reply:
Makes perfect sense.

Reply:
Done.